File: myq1.c
#include<stdio.h>
int row_number(long int n, int alpha,int beta);
char seat_position(int n);
char direction(long int n);
int main()
{
	long int t,n,i=0;
	char sea_pos,dir;
	int row_num,temp;
	scanf("%ld",&t);
	while(i < t)
	{
		i++;
		scanf("%ld",&n);
		if (n != 1)
		{
			temp = n % 10;
			sea_pos = seat_position(temp);
			row_num = row_number()
			printf("%d %c %c\n",row_num,sea_pos,dir);	
		}
		else
			printf("poor conductor\n");
	}
	return 0;
}
char seat_position(int n)
{
	if (n == 1 || n == 2 || n == 6 || n == 7)
		return 'W';
	else if (n == 3 || n == 0 || n == 4 || n == 9)
		return 'A';
	else
		return 'M';	
}

File: myq1.py
import sys
input_data = sys.stdin.readlines()
input_data.pop(0)
dictionary = {}
def odd(i):
	if i in dictionary:
		return dictionary[i]
	else:
		n = (i -2)/10 + 1
		dictionary[i] = 2*n -1
		return dictionary[i]

def even(i):
	if i in dictionary:
		return dictionary[i]
	else:
		n = (i -11)/10 + 1
		dictionary[i] = 2*n
		return dictionary[i]
		
for i in input_data:
	try:
		a = False
		i = i.rstrip()
		if i == '1':
			print "poor conductor"
			a = True
		elif i.endswith('2') or i.endswith('1'):
			seat_position = 'W'
			direction = 'L'
			if i.endswith('2'):
				i = int(i)
				row_number = odd(i)
			else:
				i = int(i)
				row_number = even(i)

		elif i.endswith('3') or i.endswith('0'):
			seat_position = 'A'
			direction = 'L'
			if i.endswith('3'):
				i = int(i)-1
				row_number = odd(i)
			else:
				i = int(i)+1
				row_number = even(i)

		elif i.endswith('4') or i.endswith('9'):
			seat_position = 'A'
			direction = 'R'
			if i.endswith('4'):
				i = int(i)-2
				row_number = odd(i)
			else:
				i = int(i)+2
				row_number = even(i)

		elif i.endswith('5') or i.endswith('8'):
			seat_position = 'M'
			direction = 'R'
			if i.endswith('5'):
				i = int(i) - 3
				row_number = odd(i)
			else:
				i = int(i) + 3
				row_number = even(i)

		elif i.endswith('6') or i.endswith('7'):
			seat_position = 'W'
			direction = 'R'
			if i.endswith('6'):
				i = int(i) - 4
				row_number = odd(i)
			else:
				i = int(i) + 4
				row_number = even(i)
		if not a:
			print row_number, seat_position, direction
	except:
		pass



			

